libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```diff
@@ -0,0 +1,207 @@
+        /// The <see cref="IStreamingTransportServer"/> this handler listens for incoming messages on.
+        /// </value>
+        public IStreamingTransportServer Server { get; set; }
+
+        /// <summary>
+        /// Processes incoming requests and returns the response, if any.
+        /// Checks the validity of the request and invokes the bot.
+        /// </summary>
+        /// <param name="request">A ReceiveRequest from the connected channel.</param>
+        /// <param name="context">Optional context to operate within. Unused in bot implementation.</param>
+        /// <param name="logger">Optional logger used to log request information and error details.</param>
+        /// <returns>A response created by the BotAdapter to be sent to the client that originated the request.</returns>
+        public override async Task<Response> ProcessRequestAsync(ReceiveRequest request, object context = null, ILogger<RequestHandler> logger = null)
+        {
+            var response = new Response();
+
+            try
+            {
+                if (request == null ||
+                    string.IsNullOrEmpty(request.Verb) ||
+                    string.IsNullOrEmpty(request.Path))
+                {
+                    response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    logger?.LogInformation("Request missing verb and/or path.");
+                }
+                else if (string.Equals(request.Verb, Request.GET, StringComparison.InvariantCultureIgnoreCase) &&
+                         string.Equals(request.Path, "/api/version", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    response.StatusCode = (int)HttpStatusCode.OK;
+                    response.SetBody(new VersionInfo() { UserAgent = userAgent });
+                }
+                else if (string.Equals(request.Verb, Request.POST, StringComparison.InvariantCultureIgnoreCase) &&
+                         string.Equals(request.Path, "/api/messages", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    var body = request.ReadBodyAsString();
+                    if (string.IsNullOrEmpty(body) || request.Streams?.Count == 0)
+                    {
+                        response.StatusCode = (int)HttpStatusCode.BadRequest;
+                        logger?.LogInformation("Request missing body and/or streams.");
+                        return response;
+                    }
+
+                    try
+                    {
+                        var adapter = new BotFrameworkStreamingExtensionsAdapter(Server, middlewareSet, logger);
+                        var bot = services?.GetService<IBot>() ?? this.bot;
+
+                        if (bot == null)
+                        {
+                            throw new Exception("Unable to find bot when processing request.");
+                        }
+
+                        if (onTurnError != null)
+                        {
```

Comment:
Use a default OnTurnError (setup in your ctor) is none was originally passed in. Get rid of this null check.

Reply:
The BotAdapter is looking for a null value to decide how to handle onturnerrors, I think a default would cause unexpected behavior.